resources/startup.sh

```diff
@@ -4,129 +4,133 @@ set -o nounset
-  -Djavax.net.ssl.trustStore=${TRUSTSTORE} \
-	-Djavax.net.ssl.trustStorePassword=changeit \
-  -Dnexus-work=/var/lib/nexus -Dnexus-webapp-context-path=/nexus \
-  -cp conf/:$(echo lib/*.jar | sed -e "s/ /:/g") \
-  org.sonatype.nexus.bootstrap.Launcher ./conf/jetty.xml ./conf/jetty-requestlog.xml"
+  # create truststore
+  TRUSTSTORE="${NEXUS_DATA_DIR}/truststore.jks"
+  create_truststore.sh "${TRUSTSTORE}" > /dev/null
 
-if ! [ -d /var/lib/nexus/plugin-repository/nexus-cas-plugin-"${CAS_PLUGIN_VERSION}" ]; then
-      echo "No cas-plugin installed"
+  echo "Setting nexus.vmoptions and properties..."
+  setNexusVmoptionsAndProperties
 
-      startNexusAndWaitForHealth
+  echo "Starting Nexus and waiting for healthy state..."
+  startNexusAndWaitForHealth ${ADMINUSER} ${ADMINDEFAULTPASSWORD}
 
-      ADMPW=$(set_random_admin_password)
+  echo "Configuring Nexus..."
+  configureNexusAtFirstStart
 
-      # add cas Plugin
-      cp -dR /opt/sonatype/nexus/resources/nexus-cas-plugin-"${CAS_PLUGIN_VERSION}"/ /var/lib/nexus/plugin-repository/
-      # add mailconfig
-      MAIL_CONFIGURATION=$(curl -s -H 'content-type:application/json' -H 'accept:application/json' 'http://127.0.0.1:8081/nexus/service/local/global_settings/current' -u "$ADMUSR":"$ADMPW" | jq ".data.smtpSettings+={\"host\": \"postfix\"}" | jq ".data.smtpSettings+={\"username\": \"\"}" | jq ".data.smtpSettings+={\"password\": \"\"}" | jq ".data.globalRestApiSettings+={\"baseUrl\": \"https://$FQDN/nexus/\"}" | jq ".data.smtpSettings+={\"systemEmailAddress\": \"${MAIL_ADDRESS}\"}" | jq ".data+={\"securityAnonymousAccessEnabled\": false}" | jq ".data+={\"securityRealms\": [\"XmlAuthenticatingRealm\",\"XmlAuthorizingRealm\",\"CasAuthenticatingRealm\"]}")
-      curl -s --retry 3 --retry-delay 10 -H "Content-Type: application/json" -X PUT -d "$MAIL_CONFIGURATION" "http://127.0.0.1:8081/nexus/service/local/global_settings/current" -u "$ADMUSR":"$ADMPW"
-      # disable new version info
-      curl -s -H "Content-Type: application/json" -X PUT -d "{data:{enabled:false}}" "http://127.0.0.1:8081/nexus/service/local/lvo_config" -u "$ADMUSR":"$ADMPW"
-      kill $!
-fi
+  echo "Stopping Nexus..."
+  stopNexus
 
-if  ! doguctl config -e "admin_password" > /dev/null ; then
-  startNexusAndWaitForHealth
-  set_random_admin_password
-  kill $!
+  doguctl config successfulInitialConfiguration true
 fi
 
-ADMPW=$(doguctl config -e "admin_password")
+echo "Getting current admin password"
+NEXUS_PASSWORD=$(doguctl config -e admin_password)
+export NEXUS_PASSWORD=${NEXUS_PASSWORD}
+
+echo "Starting Nexus and waiting for healthy state..."
+startNexusAndWaitForHealth ${ADMINUSER} ${NEXUS_PASSWORD}
 
-startNexusAndWaitForHealth
-setProxyConfiguration
-kill $!
+echo "Configuring Nexus..."
+configureNexusAtSubsequentStart
 
-echo "render_template"
-# update cas url
-render_template "/opt/sonatype/nexus/resources/cas-plugin.xml.tpl" > "/var/lib/nexus/conf/cas-plugin.xml"
+echo "Stopping Nexus..."
+stopNexus
```

Comment:
Do we need to restart nexus on every startup?

Reply:
yes, because the last command executed in the startup script has to be the ".../nexus run" command, so when the dogus is stopped, nexus will be stopped in a controlled kind of way and not killed of.